match/lib/match/git_helper.rb

```diff
@@ -79,6 +81,42 @@ def self.clear_changes
       @dir = nil
     end
 
+    # Create and checkout an specific branch in the git repo
+    def self.checkout_branch(branch)
+      return unless @dir
+
+      commands = []
+      if branch_exists?(branch)
+        commands << "git checkout #{branch}"
+      else
+        commands << "git checkout --orphan #{branch}"
+        commands << "git reset --hard"
```

Comment:
Maybe add a comment on why we do this. It's so that the newly created branch doesn't contain the certificates and profiles from the main branch, right?